change from deprecated ...set_option to set_options

Spirit-act opened the pull request:
In php8.4 stream_context_set_option no longer supports the signature with two arguments. Therefor the alternative function stream_context_set_options needs to be used.

source: https://www.php.net/manual/en/function.stream-context-set-option.php

> ## ~The Change is only slightly backwards compatible, because the new function was introduced in PHP8.3.~

Also implicitly marking parameters as nullable is deprecated.

The implicit nullable is only changed by files, which I use within my script, because only these I could validate.


### Edit:

In the last commit a check for the PHP version was added. If the PHP version is greater or equal to PHP 8.3.0 `stream_context_set_options` is used. If the version is lower `stream_context_set_option` is used.

It feels a bit ugly, but i did not find a better solution for backwards compatibility.

Spirit-act committed:
change from deprecated ...set_option to set_options

In php8.4 stream_context_set_option no longer supports the signature
with two arguments. Therefor the alternative function
stream_context_set_options needs to be used.

source: https://www.php.net/manual/en/function.stream-context-set-option.php

Also implicitly marking parameters as nullable is deprecated.

Signed-off-by: Spirit-act <16631523+Spirit-act@users.noreply.github.com>

Spirit-act committed:
add missing nullable typehint

Signed-off-by: Spirit-act <16631523+Spirit-act@users.noreply.github.com>

ghnp5 commented:
Looks like this will fix #805.

Any estimate date when this will be merged?

Thanks!!

Spirit-act committed:
resolve merge conflicts

Merge branch 'main' into soap-php8.4
- remove 202408
- add 202505
- add fixes to 202505

Signed-off-by: Spirit-act <16631523+Spirit-act@users.noreply.github.com>

Spirit-act committed:
add check for php version for backwards compatiblity

stream_context_set_options only exists since php8.3.0. This Commit
intruces a check for the php version. If the version is greater or equal
than 8.3.0 it uses stream_context_set_options. Otherwise it uses stream_context_set_option.

Signed-off-by: Spirit-act <16631523+Spirit-act@users.noreply.github.com>

Spirit-act committed:
correct php version id check

the php version id check for backwards compatibility had one
digit to much.

Signed-off-by: Spirit-act <16631523+Spirit-act@users.noreply.github.com>

Spirit-act committed:
resolve merge conflicts to v70

- remove 202411
- add 202508
- adjust LineItem and LineItemSummary

Signed-off-by: Spirit-act <16631523+Spirit-act@users.noreply.github.com>

Spirit-act committed:
fix implicit null in ApiException

Signed-off-by: Spirit-act <16631523+Spirit-act@users.noreply.github.com>

Spirit-act committed:
fix implicite null in ApiError aswell

also fix in v202411

Signed-off-by: Spirit-act <16631523+Spirit-act@users.noreply.github.com>

Spirit-act committed:
replace every occurence of implicit null type

replaced every "array $var = null"
with "?array $var = null"

Signed-off-by: Spirit-act <16631523+Spirit-act@users.noreply.github.com>

Spirit-act committed:
fix accidental double questionmarks

Signed-off-by: Spirit-act <16631523+Spirit-act@users.noreply.github.com>

Spirit-act committed:
merge upstream relase 71.0.0

Signed-off-by: Spirit-act <16631523+Spirit-act@users.noreply.github.com>

Spirit-act committed:
Merge branch 'main' into soap-php8.4

Signed-off-by: Spirit-act <16631523+Spirit-act@users.noreply.github.com>
